node/lib/util/git_util.js

```diff
@@ -911,3 +911,66 @@ exports.configIsTrue = co.wrap(function*(repo, configVar) {
     return (configured === "true" || configured === "yes" ||
             configured === "on");
 });
+
+/**
+ * Fetch the remote having the specified `repo`.
+ * Throw a `UserError` object if the repository cannot be fetched.
+ *
+ * @async
+ * @param {NodeGit.Repository} repo
+ */
+exports.getRefs = co.wrap(function *(repo) {
+	assert.instanceOf(repo, NodeGit.Repository);
+
+    const execString = `git -C '${repo.path()}' ls-remote --refs`;
+
+    try {
+        const result = yield ChildProcess.exec(execString);
+        return result.stdout.split("\n");
+    }
+    catch (e) {
+        return new UserError(e.message);
+    }
+});
+
+/**
+ * Delete the specified `ref` on the specified remote `repo`.
+ * Throw a `UserError` object if cannot push to remote repository.
+ *
+ * @async
+ * @param {NodeGit.Repository} repo
+ * @param {String}             remoteName
+ */
+exports.removeRemoteRef = co.wrap(function *(repo, ref) {
```

Comment:
Given the likely number of these, I wonder if it would make sense to do batch deletion.  That is, push multiple deletes at once.